13_BitonicSort/app_resources/bitonic_sort.hlsl

```diff
@@ -0,0 +1,129 @@
+#include "nbl/builtin/hlsl/bda/bda_accessor.hlsl"
+#include "app_resources/common.hlsl"
+
+// Define the Push Constant Structure
+[[vk::push_constant]] BitonicPushData pushData;
+
+// Define Accessor Types
+using Ptr = nbl::hlsl::bda::__ptr < uint32_t >;
+using PtrAccessor = nbl::hlsl::BdaAccessor < uint32_t >;
+
+// Define the Group Shared Memory
+groupshared uint32_t sdataKey[WorkgroupSize];
+groupshared uint32_t sdataValue[WorkgroupSize];
+
+// Define the Shared Accessor for Bitonic Sort
+struct SharedAccessor
+{
+    uint32_t get(const uint32_t index)
+    {
+        return sdataKey[index];
+    }
+
+    void set(const uint32_t index, const uint32_t value)
+    {
+        sdataKey[index] = value;
+    }
+
+    uint32_t getValue(const uint32_t index)
+    {
+        return sdataValue[index];
+    }
+
+    void setValue(const uint32_t index, const uint32_t value)
+    {
+        sdataValue[index] = value;
+    }
+
+    void workgroupExecutionAndMemoryBarrier()
+    {
+        nbl::hlsl::glsl::barrier();
+    }
+};
+
+// Bitonic Sort Implementation
+void bitonic_sort(uint32_t tid, inout PtrAccessor key, inout PtrAccessor val, inout SharedAccessor sdata, uint32_t size)
```

Comment:
see how the counting sort is done via a templated struct functor (so that static polymorphism is achieved)